src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incentives/IncentivesResourceIntTest.kt

```diff
@@ -516,4 +518,270 @@ class IncentivesResourceIntTest : IntegrationTestBase() {
       )
     }
   }
+
+  @DisplayName("get global incentive level")
+  @Nested
+  inner class getGlobalIncentiveLevel {
+    @Test
+    fun `get global incentive level`() {
+      webTestClient.get().uri("/incentives/reference-codes/STD")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCENTIVES")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("code").isEqualTo("STD")
+        .jsonPath("active").isEqualTo(true)
+    }
+
+    @Test
+    fun `get global incentive level returns 404 if IEP_LEVEL doesn't exist`() {
+      webTestClient.get().uri("/incentives/reference-codes/HHH")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCENTIVES")))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `get global incentive prevents access without appropriate role`() {
+      assertThat(
+        webTestClient.get().uri("/incentives/reference-codes/STD")
+          .headers(setAuthorisation(roles = listOf("ROLE_BLA")))
+          .exchange()
+          .expectStatus().isForbidden,
+      )
+    }
+
+    @Test
+    fun `get global incentive prevents access without authorization`() {
+      assertThat(
+        webTestClient.get().uri("/incentives/reference-codes/STD")
+          .exchange()
+          .expectStatus().isUnauthorized,
+      )
+    }
+  }
+
+  @DisplayName("create global incentive level")
+  @Nested
+  inner class createGlobalIncentiveLevel {
+
+    @AfterEach
+    internal fun deleteIncentiveLevel() {
+      webTestClient.delete().uri("/incentives/reference-codes/NIEP")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCENTIVES")))
+        .exchange()
+        .expectStatus().isOk
+    }
+
+    @Test
+    fun `create global incxvcxvcentive level`() {
+      assertThat("NIEP").isEqualTo("NIEP")
+    }
+
+    @Test
+    fun `create global incentive level`() {
+      val response = webTestClient.post().uri("/incentives/reference-codes")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCENTIVES")))
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(
+          BodyInserters.fromValue(
+            """{
+            "code"    : "NIEP",
+            "domain" : "IEP_LEVEL",
```

Comment:
is domain really in the body? Not sure if it needs to be

Reply:
no it doesn't - was reusing a dto which included it. I'll create a new dto on next PR